spring-cloud-gcp-data-datastore/src/test/java/com/google/cloud/spring/data/datastore/repository/query/GqlDatastoreQueryTests.java

```diff
@@ -447,6 +448,39 @@ public void pageableTestPageCursor() {
 				.queryKeysOrEntities(any(), eq(Trade.class));
 	}
 
+	@Test
+	public void streamResultTest() {
+		Mockito.<Class>when(this.queryMethod.getReturnedObjectType())
+				.thenReturn(Trade.class);
+		Parameters parameters = mock(Parameters.class);
+
+		when(this.queryMethod.getParameters())
+				.thenReturn(parameters);
+		when(parameters.getNumberOfParameters())
+				.thenReturn(0);
+		when(this.queryMethod.isStreamQuery()).thenReturn(true);
+
+		Trade tradeA = new Trade();
+		tradeA.id = "a";
+		Trade tradeB = new Trade();
+		tradeB.id = "b";
+		doAnswer(invocation -> {
+			GqlQuery statement = invocation.getArgument(0);
+			assertThat(statement.getQueryString()).isEqualTo("unusedGqlString");
```

Comment:
I vaguely remember something about asserts inside of `when()`/`doAnswer()` not propagating up to the test level. Could you see if replacing "unusedGqlString" with "something to trigger assert exception" makes the test fail? If it does not fail, then we might as well remove this assert line, since it will never work.

(but my recollection might be wrong! So if changing the string makes the test fail properly, then definitely leave it alone).

Reply:
Tested. It actually did fail properly!